crates/spark/src/tree/store.rs

```diff
@@ -93,16 +93,21 @@ impl TreeStore for InMemoryTreeStore {
             .collect();
 
         for (key, entry) in leaves_state.leaves_reservations.clone().iter() {
-            // remove leaves not existing in the main pool
-            let mut filtered_leaves: Vec<TreeNode> = entry
-                .leaves
-                .iter()
-                .filter(|l| {
-                    leaves_state.leaves.contains_key(&l.id)
-                        || leaves_state.missing_operators_leaves.contains_key(&l.id)
-                })
-                .cloned()
-                .collect();
+            // remove leaves not existing in the main pool, except for Swap reservations
```

Comment:
What is the difference between swaps and other reservations (like lightning send for example)?